spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -226,7 +226,12 @@ class CometSparkSessionExtensions
 
     private def isCometPlan(op: SparkPlan): Boolean = op.isInstanceOf[CometPlan]
 
-    private def isCometNative(op: SparkPlan): Boolean = op.isInstanceOf[CometNativeExec]
+    private def isCometNative(op: SparkPlan): Boolean = op match {
+      case _: CometNativeExec => true
+      case AQEShuffleReadExec(q: ShuffleQueryStageExec, _) =>
+        isCometNative(q.plan)
```

Comment:
Hmm, `ShuffleQueryStageExec`'s child must be a `ShuffleExchangeLike` or `ReusedExchangeExec`. In Comet query, it should be Comet shuffle which is `CometPlan`.

Did you see there is a `CometNativeExec` under `ShuffleQueryStageExec`?